File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int k, int[] score) {
         int length = score.Length;

                int[] answer = new int[length];
                List<int> honor = new List<int>();

                for (int i = 0; i < length; i++)
                {
                    honor.Add(score[i]);
                    honor.Sort();

                    if (i < k)
                        answer[i] = honor[0];
                    else
                        answer[i] = honor[i - k + 1];
                }
                return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []

    rank = []

    for s in score:
        if len(rank) < k:
            rank.append(s)
            rank.sort()
        else:
            if rank[0] < s:
                rank[0] = s
                rank.sort()
        answer.append(rank[0])
    return answer